bus-mapping/src/evm/opcodes/balance.rs

```diff
@@ -57,23 +57,28 @@ impl Opcode for Balance {
         // Read account balance.
         let account = state.sdb.get_account(&address).1;
```

Comment:
You can just do `let (exists, account) = state.sdb.get_account(&address);`

Reply:
I have tested this option and unfortunately it's not viable without further changes:
- Currently the flag returned by `get_account` is true when the account was found in the StateDB, but the account may be empty
- In the CircuitInputBuilder we're currently loading a bunch of empty accounts in the StateDB, so an empty account is found in the StateDB
- We're using the `found` flag as a check to make sure that all accounts that are touched during a block are preloaded into the StateDB (even if the account is empty).  This allows us to detect if we missed retrieving some account from geth that will be used in a block.  See all the cases where `Err(Error::AccountNotFound(address));` is returned.
  - If the flag `found` is replaced to mean `exists`, then we lose this check, and we will not be able to detect the case where we are dealing with an account that we didn't retrieve (so it's empty but shouldn't be).

So I think it's better to leave the flag returned by `get_account` to mean `found` (as in, the account is preloaded in StateDB even if it's empty), and use the `account.is_empty()` to really determine if the account exists or not.

Reply:
got it. it is unfortunate that it's possible to for the db find accounts that the circuits will later claim don't exist. i'll look into whether it's possible to fix this naming issue.